File: SuperService/SuperService/MessageCenter/EaseMob/EaseUI/EMUIKit/Util/Category/UIViewController+HUD.m
/************************************************************
  *  * EaseMob CONFIDENTIAL 
  * __________________ 
  * Copyright (C) 2013-2014 EaseMob Technologies. All rights reserved. 
  *  
  * NOTICE: All information contained herein is, and remains 
  * the property of EaseMob Technologies.
  * Dissemination of this information or reproduction of this material 
  * is strictly forbidden unless prior written permission is obtained
  * from EaseMob Technologies.
  */

#import "UIViewController+HUD.h"

#import "MBProgressHUD.h"
#import <objc/runtime.h>

static const void *HttpRequestHUDKey = &HttpRequestHUDKey;

@implementation UIViewController (HUD)

- (MBProgressHUD *)HUD {
    return objc_getAssociatedObject(self, HttpRequestHUDKey);
}

- (void)setHUD:(MBProgressHUD *)HUD {
    objc_setAssociatedObject(self, HttpRequestHUDKey, HUD, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
}

- (void)showHudInView:(UIView *)view hint:(NSString *)hint {
  MBProgressHUD *HUD = [[MBProgressHUD alloc] initWithView:view];
  HUD.labelText = hint;
  HUD.labelFont = [UIFont systemFontOfSize:12.0];
  [view addSubview:HUD];
  [HUD show:YES];
  [self setHUD:HUD];
}

- (void)showHint:(NSString *)hint yOffset:(float)yOffset {
  //显示提示信息
  UIView *view = [[UIApplication sharedApplication].delegate window];
  MBProgressHUD *hud = [MBProgressHUD showHUDAddedTo:view animated:YES];
  hud.userInteractionEnabled = NO;
  // Configure for text only and offset down
  hud.mode = MBProgressHUDModeText;
  hud.labelText = hint;
  hud.labelFont = [UIFont systemFontOfSize:12.0];
  hud.margin = 10.f;
  hud.yOffset = 180;
  hud.yOffset += yOffset;
  hud.removeFromSuperViewOnHide = YES;
  [hud hide:YES afterDelay:2];
}

- (void)hideHud {
  [[self HUD] hide:YES];
}

- (void)showHUDInView:(UIView *)view withLoading:(NSString *)hint {
    MBProgressHUD *HUD = [[MBProgressHUD alloc] initWithView:view];
    HUD.labelText = hint;
    HUD.labelFont = [UIFont systemFontOfSize:12.0];
    [view addSubview:HUD];
    [HUD show:YES];
    [self setHUD:HUD];
}

- (void)showHint:(NSString *)hint
{
    //显示提示信息
    UIView *view = [[UIApplication sharedApplication].delegate window];
    MBProgressHUD *HUD = [MBProgressHUD showHUDAddedTo:view animated:YES];
    HUD.userInteractionEnabled = NO;
    HUD.mode = MBProgressHUDModeText;
    HUD.labelText = hint;
    HUD.labelFont = [UIFont systemFontOfSize:12.0];
    HUD.margin = 10.f;
    HUD.removeFromSuperViewOnHide = YES;
    [HUD hide:YES afterDelay:2];
}

- (void)showHint:(NSString *)hint withFontSize:(CGFloat) fontSize
{
  //显示提示信息
  UIView *view = [[UIApplication sharedApplication].delegate window];
  MBProgressHUD *HUD = [MBProgressHUD showHUDAddedTo:view animated:YES];
  HUD.userInteractionEnabled = NO;
  HUD.mode = MBProgressHUDModeText;
  HUD.labelText = hint;
  HUD.labelFont = [UIFont systemFontOfSize:fontSize];
  HUD.margin = 10.f;
  HUD.removeFromSuperViewOnHide = YES;
  [HUD hide:YES afterDelay:2];
}

- (void)showErrorHint:(NSError *)error
{
  //显示提示信息
  [self showErrorHint:error withFontSize:12];
}

- (void)showErrorHint:(NSError *)error withFontSize:(CGFloat) fontSize
{
  //显示提示信息
  NSString *msg;
  if(error.userInfo[@"resDesc"]) {
    msg = error.userInfo[@"resDesc"];
  } else {
    msg = [NSString stringWithFormat:@"数据请求错误:%ld", error.code];
  }
  [self showHint:msg withFontSize:fontSize];
}

- (void)hideHUD {
    [[self HUD] hide:YES];
}

@end

File: SuperService/SuperService/Extension/UIViewController+HUD.swift
//
//  UIViewController+HUD.swift
//  SuperService
//
//  Created by Hanton on 11/11/15.
//  Copyright © 2015 ZKJS. All rights reserved.
//

import Foundation

extension UIViewController {

  private struct AssociatedKeys {
    static var ZKJS_HUD = "ZKJS_HUD"
  }
  
  var HUD: MBProgressHUD? {
    get {
      return objc_getAssociatedObject(self, &AssociatedKeys.ZKJS_HUD) as? MBProgressHUD
    }
    
    set {
      if let newValue = newValue {
        objc_setAssociatedObject(
          self,
          &AssociatedKeys.ZKJS_HUD,
          newValue as MBProgressHUD?,
          .OBJC_ASSOCIATION_RETAIN_NONATOMIC
        )
      }
    }
  }
  
  func showHUDInView(view: UIView, withLoading hint: String) {
    let HUD = MBProgressHUD(view: view)
    HUD.labelText = hint
    view.addSubview(HUD)
    HUD.show(true)
    self.HUD = HUD
  }
  
  func showHint(hint: String) {
    if let view = UIApplication.sharedApplication().delegate?.window {
      let HUD = MBProgressHUD.showHUDAddedTo(view, animated: true)
      HUD.userInteractionEnabled = false
      HUD.mode = .Text
      HUD.labelText = hint
      HUD.margin = 10.0
      HUD.removeFromSuperViewOnHide = true
      HUD.hide(true, afterDelay: 1)
    }
  }
  
  func hideHUD() {
    HUD?.hide(true)
  }
  
}
